tracer/src/Datadog.Trace/Iast/Ranges.cs

```diff
@@ -206,22 +207,42 @@ internal static Range[] CopyWithMark(Range[] ranges, SecureMarks secureMarks)
         return newRanges.ToArray();
     }
 
-    internal static bool ContainsUnsafeRange(IEnumerable<Range>? ranges)
+    internal static Range[]? GetUnsafeRanges(Range[] ranges, SecureMarks safeMarks, SourceType[]? safeSources)
```

Comment:
I think this always returns non null?

```suggestion
    internal static Range[] GetUnsafeRanges(Range[] ranges, SecureMarks safeMarks, SourceType[]? safeSources)
```